sympy/vector/functions.py

```diff
@@ -214,6 +214,38 @@ def gradient(scalar, coord_sys):
 
     return coord_sys.delop(scalar).doit()
 
+def directional_derivative(scalar, vect, coord_sys):
+    """
+    Returns the directional derivative of a scalar field computed along a given vector
+    in given coordinate system.
+
+    Parameters
+    ==========
+
+    scalar : SymPy Expr
+        The scalar field to compute the gradient of
+
+    vect : Vector
+        The vector operand
+
+    coord_sys : CoordSysCartesian
+        The coordinate system to calculate the gradient in
+
+    Examples
+    ========
+
+    >>> from sympy.vector import CoordSysCartesian, directional_derivative
+    >>> R = CoordSysCartesian('R')
+    >>> f1 = R.x*R.y*R.z
+    >>> v1 = 3*R.i + 4*R.j + R.k
+    >>> directional_derivative(f1, v1, R)
```

Comment:
@Upabjojr ,  I understand what you mean, but as I told you earlier it would be a very large amount of work to clean up the whole coordinate system with each function. So for the existing system this is the best possible way